### DISCUSSION TITLE
Defining `get_fs_uniquifier()` on the user class

### DISCUSSION THREAD

**Discussion by savchenko** [pallets-eco/flask-security#996] [Q&A] [answered]
I have a user model where `id` column is a primary key as well as a valid UUID. I would like to declare a method:
```py
def get_fs_uniquifier(self):
    return self.id
```
...and use preexisting column instead of the additional `fs_uniquifier`.

Should I expect any ill side-effects? Do I need to add any extra code for Flask-Login? E.g. from https://github.com/Flask-Middleware/flask-security/issues/312

```py
def install_user_loader(app):
    '''In order to handle non-standard id column name'''
    with app.app_context():
        security = app.extensions['security']
        @app.login_manager.user_loader
        def user_loader(user_id):
            return security.datastore.find_user(user_id=user_id)
```

**Comment by jwag956** [comment, marked as answer]
I would strongly suggest you not do this - main reason - you won't be able to reset any security attributes - including changing password. It also ties the existence of a user to their security attributes which is unlikely to be what users expect

**Reply by savchenko** [reply]
> you won't be able to reset any security attributes - including changing password

That's important :) Might be worth adding to the docs? Happy to draft pull request.

**Reply by savchenko** [reply]
Thinking more about it, do you reckon there might be some side-effects if the `fs_uniquifier` is made `PK` for the table?

Or is there an expectation that separate `id` exists?

**Reply by jwag956** [reply]
Same issue - various features such as reset password change the fs_uniquifier which would render your User model useless. So yes - you need a separate fs_uniqifier field that is completely managed by Flask-Security.
